src/Process/Signal.php

```diff
@@ -100,10 +110,35 @@ public function handleSignal(int $signalNumber, $signalInformation): void
                 // For now, SITGTERM is safe.
                 $this->processSignalInformation($information);
             } else {
-                $this->bufferedSignals[] = $information;
+                $this->bufferSignalInformation($information);
             }
         }
 
         return;
     }
+
+    protected function bufferSignalInformation(InformationInterface $information): SignalInterface
+    {
+        if ($this->getCanBufferSignals()) {
+            $this->bufferedSignals[] = $information;
+        }
+
+        return $this;
+    }
+
+    public function setCanBufferSignals(bool $canBufferSignals): SignalInterface
+    {
+        if ($this->canBufferSignals !== $canBufferSignals) {
+            $this->canBufferSignals = $canBufferSignals;
+        } else {
+            throw new \LogicException(sprintf('Can buffer signals is already set to [%s]', $this->canBufferSignals));
```

Comment:
This error message might look a bit weird:

```
php > var_dump(sprintf('%s OR %s', true, false));
string(5) "1 OR "

Reply:
good catch thanks